lib/chef/mixin/windows_architecture_helper.rb

```diff
@@ -42,6 +36,16 @@ def wow64_architecture_override_required?(node, desired_architecture)
           is_i386_process_on_x86_64_windows?
       end
 
+      def forced_32bit_override_required?(node, desired_architecture)
+        desired_architecture == :i386 &&
+          node_windows_architecture(node) == :x86_64 &&
+          !is_i386_process_on_x86_64_windows?
+      end
+
+      def wow64_directory
+        Chef::ReservedNames::Win32::System.get_system_wow64_directory
+      end
+
       def with_os_architecture(node, architecture: nil)
```

Comment:
I'm a bit worried that with_os_architecture is left unmodified.  Both lib/chef/util/powershell/cmdlet.rb and lib/chef/mixin/powershell_out.rb  uses it to launch commands in the right architecture and this won't handle our 64/32 conversion.  It only fixes the providers.  I feel like we should perhaps refactor and re-architect all users of this mixin to not use with_os_architecture with its global redirection hammer and instead introduce a "get_system_dir(node, architecture: nil)" method which returns the correct prefix.  That prefix can then be used by each of these classes to correctly launch powershell, or regedit or cmd whatever it is they want to launch.